applications/DEMApplication/custom_strategies/strategies/explicit_solver_strategy.cpp

```diff
@@ -913,54 +915,66 @@ namespace Kratos {
 
         KRATOS_TRY
         ClearFEMForces();
-        ConditionsArrayType& rConditions = GetFemModelPart().GetCommunicator().LocalMesh().Conditions();
-        ProcessInfo& r_process_info = GetFemModelPart().GetProcessInfo();
-        const ProcessInfo& r_const_process_info = GetFemModelPart().GetProcessInfo();
 
+        for (ModelPart::SubModelPartsContainerType::iterator sub_model_part = GetFemModelPart().SubModelPartsBegin(); sub_model_part != GetFemModelPart().SubModelPartsEnd(); ++sub_model_part) {
```

Comment:
behold the power of `auto` :D

```suggestion
        for (auto& submp : GetFemModelPart().SubModelParts()) {
```